state-cleanup/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+use near_sdk::base64;
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::serde_json;
+use near_sdk::{env, near_bindgen};
+
+#[allow(dead_code)]
+#[near_bindgen]
+struct Contract {}
```

Comment:
Should just be able to delete this? Or am I forgetting about some dependency from v3 or otherwise?

Reply:
agree i ran this code on version `4.0.0-pre.2`
```rust
use near_sdk::serde::Deserialize;
use near_sdk::serde_json;
use near_sdk::env;
use near_sdk::json_types::Base64VecU8;

#[derive(Deserialize)]
#[serde(crate = "near_sdk::serde")]
struct Args {
    pub keys: Vec<Base64VecU8>,
}

#[no_mangle]
pub extern "C" fn clean() {
    env::setup_panic_hook();
    let input = env::input().unwrap();
    let args: Args = serde_json::from_slice(&input).unwrap();
    for key in args.keys.iter() {
        env::storage_remove(&key.0);
    }
}
```

Reply:
and `3.1.0`

Reply:
FYI, I have no idea why that might be necessary

Reply:
it helped me to solve the error: 

> The collection is an inconsistent state. Did previous smart contract execution terminate unexpectedly?

to be more precise it helps to better understand how storage works.
I left a note for myself, it may be useful not only for me https://gist.github.com/ilyar/19bdc04d1aa09ae0fc84eb4297df1a1d#file-clear_state_of_contract_on_near_protocol-md